### PR TITLE
Integrate with Flowstar.jl

### PR DESCRIPTION
**Pull request by mforets**
Closes https://github.com/JuliaReach/ReachabilityAnalysis.jl/issues/640.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -32,6 +32,7 @@ DynamicPolynomials = "0.3 - 0.4"
 Expokit = "0.2"
 ExponentialUtilities = "1"
 ExprTools = "0.1"
+Flowstar = "0.1"
```
Comment (agerlach) [inline Project.toml:35]:
```suggestion
Flowstar = "0.2"
```

File: src/Algorithms/FLOWSTAR/post.jl [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    model = if !haskey(kwargs, :model)
+        # temporary
+        throw(ArgumentError("the model file needs to be passed as a keyword argument"))
+    else
+        kwargs[:model]
+    end
+
+    # call Flow*
+    sol = FlowstarContinuousSolution(model)
+    flow = sol.flow
```
[the remark this answers is not in the document]
Comment (mforets) [reply src/Algorithms/FLOWSTAR/post.jl:17]: The dot is used when loading optional packages (Requires). Here I see some name clashes, for example:

```julia
julia> using ReachabilityAnalysis, Flowstar

julia> flowpipe
WARNING: both Flowstar and ReachabilityAnalysis export "flowpipe"; uses of it in module Main must be qualified
ERROR: UndefVarError: flowpipe not defined
```
In principle we don't need the getter functions (flowpipe) to be exported from Flowstar.jl

**Comment by mforets** [comment]
I think this is good as a first iteration; we can incorporate the commented test in another PR.

File: src/Algorithms/FLOWSTAR/post.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (mforets) [reply src/Algorithms/FLOWSTAR/post.jl:17]: For now I'm working such problem around by use of `import` instead of `using`.

**Comment by mforets** [comment]
Ok in the last commit I've added the code for model generation. I'm getting an error, not sure if this is related to a corrupt model or sth else.

**Comment by mforets** [comment]
Should be fixed now. From my side this is ready to merge.

File: src/Algorithms/FLOWSTAR/post.jl [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        # initial set
+        X0 = initial_state(ivp)
+        dom = isa(X0, IntervalBox) ? X0 : convert(IntervalBox, overapproximate(X0, Hyperrectangle))
+
+        name = "flowstar" # TODO parse name of f
+
+        # Plot first two variables if possible, otherwise plot the first variable vs time
+        # TODO disable this option
+        plot_states = n > 1 ? "$(fstr[1][1]),$(fstr[2][1])" : "t,$(fstr[1][1])" # seems mandatory?
+        gen_plots = false # ignored?
```
Comment (agerlach) [inline src/Algorithms/FLOWSTAR/post.jl:52]: FYI, in conversation with Chen he mentioned that to turn plotting off you need to prepend the `output` setting as `tm ouput` *however* a valid plot setting is still required at parse time. That is why `plot_states` is requred.

File: src/Algorithms/FLOWSTAR/post.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (agerlach) [reply src/Algorithms/FLOWSTAR/post.jl:70]: The reason to use the getter is b/c the `struct` definition could change down the line, but not be a breaking change.

**Comment by agerlach** [comment]
Looks good to me

File: src/Algorithms/FLOWSTAR/post.jl (hunk 3 above)
Comment (mforets) [reply src/Algorithms/FLOWSTAR/post.jl:52]: Ok, thanks for the clarification.

**Comment by mforets** [comment]
I got this error when trying to register the new version of RA: https://github.com/JuliaReach/ReachabilityAnalysis.jl/issues/111#issuecomment-1168126764

**Comment by schillic** [comment]
> I got this error when trying to register the new version of RA: [#111 (comment)](https://github.com/JuliaReach/ReachabilityAnalysis.jl/issues/111#issuecomment-1168126764)

This should be fixed now because the package got released. Maybe we should require v0.2.4 as the minimum version, but otherwise please try again.

**Pull request merged**